### PR TITLE
feat: refactor-index

### PR DESCRIPTION
**Pull request by oivan-plenty**
## Why:  to improve readability and for separation of logic 

Closes: #133411

## Describe your changes
index.vue was refactored, logic is now in an composable
## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

**Review by aoltean-plenty — DISMISSED** [review]

**Review by clincu-plenty — DISMISSED** [review]

File: apps/web/composables/useHomepageData/types.ts
Comment (ksted) [inline apps/web/composables/useHomepageData/types.ts:1]: **issue**: Please keep `MediaCard` types next to the component to stay consistent with the way other components are handled. The same could probably be said about `HeroItem`.

File: apps/web/composables/useHomepageData/types.ts [hunk 1]
```diff
@@ -22,7 +22,9 @@ export type HeroItem = {
   actualBackgroundSize: SizeKey;
 };
 
-export type HeroCarouselProps = {
-  background: { image: string; sizes: Sizes };
-  hero: HeroItem[];
+export type MediaItem = {
```
Comment (ksted) [inline apps/web/composables/useHomepageData/types.ts]:
```suggestion
export type MediaCardProps = {
```
All props types should have a `Props` suffix. Applies to the `HeroItem`(which should probably be `HeroContentProps` now) as well.

**Review by ksted — CHANGES_REQUESTED** [review]

File: apps/web/components/ui/HeroCarousel/types.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ export type SizeKey = 'lg' | 'md' | 'sm' | 'xs';
 
 export type Sizes = Record<SizeKey, { width: string; height: string }>;
 
-export type HeroItem = {
+export type HeroContentProps = {
   image?: string;
```
Comment (ksted) [inline apps/web/components/ui/HeroCarousel/types.ts:11]: **question**: Don't we still have three sizes here?

Comment (oivan-plenty) [reply apps/web/components/ui/HeroCarousel/types.ts:11]: yes, and the type reflects that. it comes as a lg, md, sm or xs

Comment (ksted) [reply apps/web/components/ui/HeroCarousel/types.ts:11]: Oh I didn't see it down there, sorry.

**Review by ksted — APPROVED** [review]

**Pull request merged**